streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -1513,7 +1513,7 @@ private Thread shutdownHelper(final boolean error, final long timeoutMs, final b
             }
 
             stateDirectory.close();
-            adminClient.close();
+            adminClient.close(Duration.ofMillis(timeoutMs));
```

Comment:
Query: Shouldn't we account the time already spent in closing processes i.e. timeoutMs passed to different applications might not be just constant rather decreasing? I have seen decreasing time approach in KafkaConsumer: https://github.com/apache/kafka/blob/3a0c83e49030074271dd34c20fa99baad6270c93/clients/src/main/java/org/apache/kafka/clients/consumer/internals/LegacyKafkaConsumer.java#L1141

Reply:
Well, yes, but not in this PR I guess?

For KafkaStreams, it's a general issue that we get a single timeout, but we make many calls which all take timeout parameters and thus, I would rather fix it across the board -- we did have some discussions about _how_ we would want to do it, but did not agree on a solution in the past.

What you propose is for sure one possible solution, but some people did not favor this approach, as it cuts down the timeout for later `close()` calls.

Cf https://cwiki.apache.org/confluence/display/KAFKA/KIP-459%3A+Improve+KafkaStreams%23close

Reply:
Thanks, make sense. I ll read the KIP.